include/PluginPinConnector.h

```diff
@@ -0,0 +1,158 @@
+/*
+ * PluginPinConnector.h - Specifies how to route audio channels
+ *                        in and out of a plugin.
+ *
+ * Copyright (c) 2024 Dalton Messmer <messmer.dalton/at/gmail.com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#ifndef LMMS_PLUGIN_PIN_CONNECTOR_H
+#define LMMS_PLUGIN_PIN_CONNECTOR_H
+
+#include <vector>
+
+#include "AutomatableModel.h"
+#include "lmms_export.h"
+#include "SampleFrame.h"
+#include "SerializingObject.h"
+
+class QWidget;
+
+namespace lmms
+{
+
+namespace gui
+{
+
+class PluginPinConnectorView;
+
+} // namespace gui
+
+//! Configure channel routing for a plugin's mono/stereo in/out ports
+class LMMS_EXPORT PluginPinConnector
+	: public Model
+	, public SerializingObject
+{
+	Q_OBJECT
+
+public:
+	//! [LMMS track channel][plugin channel]
+	using PinMap = std::vector<std::vector<BoolModel*>>;
+
+	struct Matrix
+	{
+		explicit Matrix(bool isIn) : in{isIn} {}
+
+		auto channelName(int channel) const -> QString;
+
+		auto enabled(std::uint8_t trackChannel, unsigned pluginChannel) const -> bool
+		{
+			return pins[trackChannel][pluginChannel]->value();
+		}
+
+		PinMap pins;
+		int channelCount = 0;
```

Comment:
I just like to initialize member variables at the point they are declared so that I don't accidentally leave them in an uninitialized state. The compiler generated default constructor is all I need here.

And `channelCount` can't be constant since eventually I'd like to implement [CLAP's audio-ports-config extension](https://github.com/free-audio/clap/blob/main/include/clap/ext/audio-ports-config.h) which allows the number of plugin channels to change. I've also written the `setTrackChannelCount` and `setPluginChannelCount` methods with that in mind - both the number of track channels and the number of plugin in/out channels should be freely changeable.